compiler/optimizer/VPHandlers.cpp

```diff
@@ -5704,6 +5704,16 @@ TR::Node *constrainAcall(TR::ValuePropagation *vp, TR::Node *node)
             TR::VPResolvedClass *newTypeConstraint = NULL;
             if (constraint)
                {
+               // Do nothing if the class of the object doesn't implement Cloneable
+               if (constraint->getClass() && !vp->comp()->fej9()->isCloneable(constraint->getClass()))
```

Comment:
The transformation is only done when the constraint is a FixedClass constraint or a ResolvedClass constraint, which ensures constraint->getClass() returns non-NULL.